src/Controllers/AmazonPayControllerReinit.php

```diff
@@ -0,0 +1,297 @@
+<?php
+
+namespace Payone\Controllers;
+
+
+use Payone\Adapter\Logger;
+use Payone\Adapter\SessionStorage;
+use Payone\Helpers\OrderHelper;
+use Payone\Helpers\PaymentHelper;
+use Payone\Methods\PayoneAmazonPayPaymentMethod;
+use Payone\Models\Api\GenericPayment\GetConfigurationResponse;
+use Payone\Models\Api\GenericPayment\GetOrderReferenceDetailsResponse;
+use Payone\PluginConstants;
+use Payone\Providers\Api\Request\GenericPaymentDataProvider;
+use Payone\Providers\Api\Request\Models\GenericPayment;
+use Payone\Services\AmazonPayService;
+use Payone\Services\Api;
+use Plenty\Modules\Order\Address\Contracts\OrderAddressRepositoryContract;
+use Plenty\Modules\Webshop\Contracts\LocalizationRepositoryContract;
+use Plenty\Plugin\Controller;
+use Plenty\Plugin\Http\Request;
+use Plenty\Plugin\Http\Response;
+use Plenty\Plugin\Templates\Twig;
+
+class AmazonPayControllerReinit extends AmazonPayController
+{
+
+    /** @var Api */
+    private $api;
+
+    /** @var GenericPaymentDataProvider */
+    private $dataProvider;
+
+    /** @var Logger */
+    private $logger;
+
+    /** @var OrderHelper */
+    private $orderHelper;
+
+    /**
+     * @param Api $api
+     * @param GenericPaymentDataProvider $dataProvider
+     * @param Logger $logger
+     * @param OrderHelper $orderHelper
+     */
+    public function __construct(Api                        $api,
+                                GenericPaymentDataProvider $dataProvider,
+                                Logger                     $logger,
+                                OrderHelper                $orderHelper
+    )
+    {
+        $this->api = $api;
+        $this->dataProvider = $dataProvider;
+        $this->logger = $logger;
+        $this->orderHelper = $orderHelper;
+    }
+
+    /**
+     * @param Twig $twig
+     * @param Response $response
+     * @param SessionStorage $sessionStorage
+     * @param $orderId
+     * @param PaymentHelper $paymentHelper
+     * @param OrderHelper $orderHelper
+     * @return string|\Symfony\Component\HttpFoundation\Response
+     * @throws \Throwable
+     * @throws \Twig_Error_Loader
+     * @throws \Twig_Error_Runtime
+     * @throws \Twig_Error_Syntax
+     */
+    public function getAmazonPayLoginWidgetReinit(Twig           $twig,
+                                                  Response       $response,
+                                                  SessionStorage $sessionStorage,
+                                                                 $orderId,
```

Comment:
Please add a valid type hint here, also for the annotations. Valid should be only INT